pkg/plugins/shell/source.go

```diff
@@ -0,0 +1,29 @@
+package shell
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/updatecli/updatecli/pkg/core/result"
+)
+
+// Source returns the stdout of the shell command if its exit code is 0
+// otherwise an error is returned with the content of stderr
+func (s *Shell) Source(workingDir string) (string, error) {
+	cmdResult, err := s.executor.ExecuteCommand(command{
+		Cmd: s.spec.Command,
+		Dir: workingDir,
+	})
+
+	if err != nil {
+		return "", err
+	}
+
+	if cmdResult.ExitCode != 0 {
+		return "", fmt.Errorf("%v The shell 🐚 command %q failed with the following message: \nstderr=\n%v\nstdout=\n%v\n", result.FAILURE, s.spec.Command, cmdResult.Stderr, cmdResult.Stdout)
```

Comment:
I do believe that a typed error could save you a whole lot of formatting duplication here :)

```
// Could be private
type ShellError struct {
    ErrCode int
    Stdout string
    Stderr string
}

func  (e *ShellError) Error() string {
    return fmt.Sprintf("yourformatted error %v%v %q", e.code, e.stdout, e.stderr)
}
```

So you can use it here

```suggestion
		return "",  &ShellError{Code: code, Stdout: stdout, Stderr: stderr}
```